schematic/schemas/data_model_jsonld.py

```diff
@@ -0,0 +1,284 @@
+        
+        # Clean up template
+        template = self.clean_template(template=template,
+                                       data_model_relationships=data_model_relationships,
+                                       )
+        # Reorder lists based on weights:
+        template = self.reorder_template_entries(template=template,)
+
+        # Add contexts to certain values
+        template = self.add_contexts_to_entries(template=template,)
+
+        return template
+
+    def add_contexts_to_entries(self, template:dict) -> dict:
+        """
+        Args:
+            template, dict: JSONLD template that has been filled up to the current node, with information
+        Returns:
+            template, dict: JSONLD template where contexts have been added back to certain values.
+        Note: This will likely need to be modified when Contexts are truly added to the model
+        """
+        for jsonld_key, entry in template.items():
+            try:
+                # Retrieve the relationships key using the jsonld_key
+                key= [k for k, v in self.rel_dict.items() if jsonld_key == v['jsonld_key']][0]
+            except:
+                continue
+            # If the current relationship can be defined with a 'node_attr_dict'
+            if 'node_attr_dict' in self.rel_dict[key].keys():
+                try:
+                    # if possible pull standard function to get node information
+                    rel_func = self.rel_dict[key]['node_attr_dict']['standard']
+                except:
+                    # if not pull default function to get node information
+                    rel_func = self.rel_dict[key]['node_attr_dict']['default']
+
+                # Add appropritae contexts that have been removed in previous steps (for JSONLD) or did not exist to begin with (csv)
+                if key == 'id' and rel_func == get_label_from_display_name:
+                    template[jsonld_key] = 'bts:' + template[jsonld_key]
+                elif key == 'required' and rel_func == convert_bool_to_str:
+                    template[jsonld_key] = 'sms:' + str(template[jsonld_key]).lower()
+        return template
+
+    def clean_template(self, template: dict, data_model_relationships: dict) -> dict:
+        '''Get rid of empty k:v pairs. Fill with a default if specified in the relationships dictionary.
+        Args:
+            template, dict: JSONLD template for a single entry, keys specified in property and class templates.
+            data_model_relationships, dict: dictionary containing information for each relationship type supported.
+        Returns:
+            template: JSONLD template where unfilled entries have been removed, or filled with default depending on specifications in the relationships dictionary.
+        '''
+        for rels in data_model_relationships.values():
+            if rels['jsonld_key'] in template.keys() and not template[rels['jsonld_key']]:
```

Comment:
here, I found `template[rels['jsonld_key']]` difficult to understand. Could you do something like: 
```
XXX = template[rels['jsonld_key']]
```
and clarify what is  `template[rels['jsonld_key']]` in a comment? 

Thank you!

Reply:
Since template is mutable, I don't want to make another copy when not necessary, but I added additional docstrings, and converted `rels['jsonld_key']` to a variable that is more descriptive.